incremental.go

```diff
@@ -196,3 +198,168 @@ func (p *IncrementalProducer) commitManifest(m *manifest, wopt *WriterOptions) e
 	}
 	return writer.Commit()
 }
+
+// ------------------------------------------------------------------------------------------
+
+// IncrementalConsumeFunc is a parsing callback which is run by the consumer every sync interval.
+type IncrementalConsumeFunc func(*ReaderIter) (data interface{}, err error)
```

Comment:
I don't like that much, why do we need this complication, there is no need for a `IncrementalConsumeFunc`, it should be just a normal `ConsumeFunc`, we just need a `Reader` that is able to transparently iterate over files